graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -869,6 +865,50 @@ def _validate_all_tags_are_used(metadata):
         )
 
 
+def _validate_output_types(outputs: Generator[Tuple[str, OutputInfo], None, None]):
```

Comment:
In cases like this consider `Iterable[Tuple[str, OutputInfo]]` as an alternative annotation. It's 100% equivalent from a type point of view, just might be a bit friendlier on the eyes since it doesn't end in `None, None`.